gpii/node_modules/testing/src/Integration.js

```diff
@@ -371,3 +371,60 @@ fluid.defaults("gpii.test.integration.mockSettingsHandlerRegistry.android", {
         }
     }
 });
+
+fluid.defaults("gpii.test.integration.deviceReporterAware", {
+    gradeNames: ["fluid.eventedComponent", "autoInit"],
+    members: {
+        deviceReporters: {}
+    },
+    listeners: {
+        onCreate: "gpii.test.integration.deviceReporterAware.populate"
+    }
+});
+
+gpii.test.integration.deviceReporterAware.mockDeviceReporters = {};
+
+gpii.test.integration.deviceReporterAware.mockDeviceReporters.findNameInList = function (expectInstalled, name) {
+    return expectInstalled.some(function (el) { return el === name; });
+};
+
+gpii.test.integration.deviceReporterAware.populate = function (that) {
+    fluid.each(that.options.deviceReporters, function (options, key) {
+        if (fluid.getGlobalValue(key)) {
+            return;
+        }
+        var expectInstalled = [];
+        fluid.each(gpii.tests.deviceReporterAware, function (tests) {
```

Comment:
This is still bizarre - why are we looping over all of the global tests, rather than the options for the test which is currently running?